src/engine/api/json_factory.cpp

```diff
@@ -233,6 +233,7 @@ util::json::Object makeRouteStep(guidance::RouteStep step, util::json::Value geo
     route_step.values["distance"] = std::round(step.distance * 10) / 10.;
     route_step.values["duration"] = std::round(step.duration * 10) / 10.;
     route_step.values["name"] = std::move(step.name);
+    route_step.values["ref"] = std::move(step.ref);
```

Comment:
Should the `ref` field always be there and empty if there is no ref for the way? Or --- similar to pronunciation and destinations --- should the ref field be undefined and only be available if there is a ref on the way?

(The same applies to the name field: shouldn't it really be optional? Or why do we separate between how we handle non-existent names and non-existent destinations)